src/scales/scale.logarithmic.js

```diff
@@ -68,96 +68,90 @@ function nonNegativeOrDefault(value, defaultValue) {
 	return helpers.isFinite(value) && value >= 0 ? value : defaultValue;
 }
 
+function getStackedValues(scale, metas) {
+	var valuesPerStack = {};
+	var meta, i, ilen, j, jlen, key, values, value;
+
+	for (i = 0, ilen = metas.length; i < ilen; ++i) {
+		meta = metas[i];
+		key = scale._getStackKey(meta);
+
+		values = valuesPerStack[key] || (valuesPerStack[key] = []);
+
+		for (j = 0, jlen = meta.data.length; j < jlen; ++j) {
+			if (meta.data[j].hidden) {
+				continue;
+			}
+
+			value = scale._getParsedValue(j, meta._datasetIndex);
+
+			if (isNaN(value) || value < 0) {
+				continue;
+			}
+
+			values[j] = values[j] || 0;
+			values[j] += value;
+		}
+	}
+	return valuesPerStack;
+}
+
+function updateLimits(scale, metas) {
+	var i, ilen, j, jlen, meta, value;
+	for (i = 0, ilen = metas.length; i < ilen; ++i) {
+		meta = metas[i];
+
+		for (j = 0, jlen = meta.data.length; j < jlen; ++j) {
+			if (meta.data[j].hidden) {
+				continue;
+			}
+
+			value = scale._getParsedValue(j, meta._datasetIndex);
+
+			if (isNaN(value) || value < 0) {
+				continue;
+			}
+
+			scale.min = Math.min(value, scale.min);
+			scale.max = Math.max(value, scale.max);
+
+			if (value !== 0) {
+				scale.minNotZero = Math.min(value, scale.minNotZero);
+			}
+		}
+	}
+}
+
 module.exports = Scale.extend({
+	_parse: function(raw) {
```

Comment:
I think there's nothing different here from the `linearBase` implementation and so it could be removed

Reply:
Sadly not based on `linearBase`

Reply:
Borrowed it, do you think that is ok?